hledger-lib/Hledger/Data/Amount.hs

```diff
@@ -961,16 +961,16 @@ accumulateMixedAmount :: AmountAccumulator -> MixedAmount -> AmountAccumulator
 accumulateMixedAmount (l, mp) ma = (l + 1, mergeAmounts r)
   where
     r = foldl' sumAmounts mempty $ amounts ma
-    mergeAmounts = M.mapWithKey (padAmounts r) . foldl' (addAmounts l) mp
+    mergeAmounts = M.mapWithKey padAmounts . foldl' addAmounts mp
     sumAmounts mp am = M.insertWith (+) (acommodity am) am mp
 
-    addAmounts l mp am
+    addAmounts mp am
       | T.null c = mp
       | c `M.member` mp = M.adjust ((:) am) c mp
       | otherwise = M.insert c (am : replicate l 0{acommodity = c}) mp
       where c = acommodity am
 
-    padAmounts r k a = if k `M.member` r then a else 0{acommodity = k} : a
+    padAmounts k a = if k `M.member` r then a else 0{acommodity = k} : a
```

Comment:
We're trying to reduce the use of the `Num` instances for `Amount`. It would probably be clearer to use `amountWithCommodity k nullamt` instead of `0{acommodity = k}`.